boards/risc-v/litex/arty_a7/src/arty_a7.h

```diff
@@ -27,6 +27,113 @@
+ * configuration
+ */
+
+#define HAVE_SDMMC          1
+#define HAVE_AUTOMOUNTER    1
+
+/****************************************************************************
+ * SDIO Configuration
+ */
+
+#define SDIO_MINOR CONFIG_NSH_MMCSDMINOR
+#define SDIO_SLOTNO CONFIG_NSH_MMCSDSLOTNO
+
+/* Can't support MMC/SD if the card interface(s) are not enable */
+
+#if !defined(CONFIG_LITEX_SDIO) && !defined(CONFIG_LITEX_SDIO1)
+#  undef HAVE_SDMMC
+#endif
+
+#if !defined(CONFIG_FS_AUTOMOUNTER)
+#  undef HAVE_AUTOMOUNTER
+#endif
+
+/****************************************************************************
+ * PROC File System Configuration
+ */
+
+#ifdef CONFIG_FS_PROCFS
+#  ifdef CONFIG_NSH_PROC_MOUNTPOINT
+#    define LITEX_PROCFS_MOUNTPOINT CONFIG_NSH_PROC_MOUNTPOINT
+#  else
+#    define LITEX_PROCFS_MOUNTPOINT "/proc"
+#  endif
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: litex_bringup
+ *
+ * Description:
+ *   Bring up board features
+ *
+ ****************************************************************************/
+
 int litex_bringup(void);
 
+/****************************************************************************
+ * Name: litex_sdio_initialize
+ *
+ * Description:
+ *   Initialize SDIO-based MMC/SD card support
+ *
+ ****************************************************************************/
+
+#ifdef HAVE_SDMMC
+int litex_sdio_initialize(void);
+#endif
+
+/****************************************************************************
+ * Name:  litex_automount_initialize
+ *
+ * Description:
+ *   Configure auto-mounters for each enabled MikroBus MMCSD
+ *
+ * Input Parameters:
+ *   None
+ *
+ *  Returned Value:
+ *    None
+ *
+ ****************************************************************************/
+
+#ifdef HAVE_AUTOMOUNTER
+int litex_automount_initialize(void);
+#endif
+
+/****************************************************************************
+ * Name: litex_automount_event
+ *
+ * Description:
+ *   The MMCSD card detection logic has detected an insertion or removal
+ *   event.  It has already scheduled the MMC/SD block driver operations.
+ *   Now we need to schedule the auto-mount event which will occur with a
+ *   substantial delay to make sure that everything has settle down.
+ *
+ * Input Parameters:
+ *   slotno - Identifies the MB slot: MB1_MMCSD_SLOTNO or MB2_MMCSD_SLOTNO.
+ *   inserted - True if the card is inserted in the slot.  False otherwise.
+ *
+ *  Returned Value:
+ *    None
+ *
+ *  Assumptions:
+ *    Interrupts are disabled.
+ *
+ ****************************************************************************/
+
+#ifdef HAVE_AUTOMOUNTER
+void litex_automount_event(int slotno, bool inserted);
+#endif
+
+#ifdef HAVE_SDMMC
+bool litex_cardinserted(int slotno);
```

Comment:
do we need a comment description for this function?

Reply:
Added